presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java

```diff
@@ -419,9 +518,18 @@ else if (!groupByColumns.isEmpty()) {
     public LinkedHashMap<VariableReferenceExpression, PinotColumnHandle> getAssignments()
     {
         LinkedHashMap<VariableReferenceExpression, PinotColumnHandle> result = new LinkedHashMap<>();
-        selections.entrySet().stream().filter(e -> !hiddenColumnSet.contains(e.getKey())).forEach(entry -> {
-            VariableReferenceExpression variable = entry.getKey();
-            Selection selection = entry.getValue();
+        if (groupByColumns != null) {
+            groupByColumns.stream().forEach(variable -> {
+                Selection selection = selections.get(variable);
+                PinotColumnHandle handle = selection.getOrigin() == Origin.TABLE_COLUMN ? new PinotColumnHandle(selection.getDefinition(), variable.getType(), PinotColumnHandle.PinotColumnType.REGULAR) : new PinotColumnHandle(variable, PinotColumnHandle.PinotColumnType.DERIVED);
+                result.put(variable, handle);
```

Comment:
Is it possible to dedup these lines with the one below [532-534] ? Particularly line 524 that creates the PinotColumnHandle. 

One possible way would be to do this in two steps: In the first step you partition outputs into: groupByColumns and those that are not in groupByColumns and not hidden. Then you can stitch these two partitions and create the column handles in one pass from them. Not sure how much code it would save, but it might help simplify the logic.

Reply:
composite the pql outputs above the stream() method.